## Data Science & AI
### NumPy (NumPy kutubxonasi)

In [1]:
#NumPy kutubxonasini chaqirish
import numpy as np

Python List & array

In [20]:
my_list = list(range(100)) # python list 0~99--> Normal
my_array = np.array(range(100)) # numpy array 0~99--> Vector

In [21]:
print(my_list)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99]


In [19]:
# my_listni run qilingandagi sarflanadigan vaqti
%time for _ in range(10): [x*2 for x in my_list]

CPU times: user 61 µs, sys: 1e+03 ns, total: 62 µs
Wall time: 66 µs


In [22]:
print(my_array)

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47
 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71
 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95
 96 97 98 99]


In [23]:
%time for _ in range(10): my_array*2

CPU times: user 34 µs, sys: 8 µs, total: 42 µs
Wall time: 45.1 µs


↑yuqirida ko'rganimiz ikkala arrayni 
xisoblashda ketadigan vaqt:

Normal: Wall time: 66 µs & 
Vector: Wall time: 45.1 µs 